### PR TITLE
logging: Fix possible out-of-bound access in log_output

### LINKED ISSUE #8999: [Coverity CID :187067] Memory - corruptions in /subsys/logging/log_output.c
Static code scan issues seen in File: /subsys/logging/log_output.c
 Category: Memory - corruptions
 Function: std_print
 Component: Other
 CID: 187067
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

### PR DESCRIPTION
**Pull request by nordic-krch**
Coverity issue #187067

Fixes #8999 
Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

**Review by ioannisg — CHANGES_REQUESTED** [review]
Looks OK to me.
I think you might need to update the documentation of the function in the header file, to stress that 0 is returned for out-of-bounds-argument, thus, it is not a valid return argument value.

Does this affect the callers of log_msg_arg_get?

File: subsys/logging/log_msg.c [hunk 1]
```diff
@@ -79,6 +79,11 @@ u32_t log_msg_arg_get(struct log_msg *msg, u32_t arg_idx)
 {
 	u32_t arg;
 
+	/* Returns 0 if requested argument not present in the message. */
```
[the remark this answers is not in the document]
Comment (nordic-krch) [reply subsys/logging/log_msg.c]: @pfalcon @ioannisg comment updated, added info to doxygen.

**Comment by nordic-krch** [comment]
@ioannisg regarding
> Does this affect the callers of log_msg_arg_get?

This API is somehow internal as it will be used only by logger backends. So unless someone adds custom logger backend it will not be used by the application.

File: subsys/logging/log_msg.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (pfalcon) [reply subsys/logging/log_msg.c]: Thanks!

**Review by pfalcon — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by nordic-krch** [comment]
@ioannisg right, missed that, rebased now.

**Review by ioannisg — APPROVED** [review]

**Pull request merged**